litestar/middleware/session/client_side.py

```diff
@@ -107,19 +118,24 @@ def get_cookie_keys(self, connection: ASGIConnection) -> list[str]:
         """
         return sorted(key for key in connection.cookies if self.cookie_re.fullmatch(key))
 
+    def get_cookie_key_set(self, connection: ASGIConnection) -> set[str]:
+        """Return a set of cookie-keys from the connection if they match the session-cookie pattern.
+
+        Args:
+            connection: An ASGIConnection instance
+
+        Returns:
+            A set of session-cookie keys
+        """
+        return {key for key in connection.cookies if self.cookie_re.fullmatch(key)}
+
     def _create_session_cookies(self, data: list[bytes], cookie_params: dict[str, Any] | None = None) -> list[Cookie]:
         """Create a list of cookies containing the session data.
         If the data is split into multiple cookies, the key will be of the format ``session-{segment number}``,
         however if only one cookie is needed, the key will be ``session``.
         """
         if cookie_params is None:
-            cookie_params = dict(
-                extract_dataclass_items(
-                    self.config,
-                    exclude_none=True,
-                    include={f for f in Cookie.__dict__ if f not in ("key", "secret")},
-                )
-            )
+            cookie_params = self._set_cookie_params
```

Comment:
```suggestion
            cookie_params = cookie_params or self._set_cookie_params
```
save a line?

Reply:
ended up removing the parameter altogether - its a private method and we only use it in one place :shrug: